src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs

```diff
@@ -13,21 +13,55 @@ namespace Raven.Client.Documents.Commands.MultiGet
 {
     public class MultiGetCommand : RavenCommand<List<GetResponse>>
     {
+        private readonly RequestExecutor _requestExecutor;
         private readonly HttpCache _cache;
         private readonly List<GetRequest> _commands;
 
         private string _baseUrl;
 
-        public MultiGetCommand(HttpCache cache, List<GetRequest> commands)
+        public MultiGetCommand(RequestExecutor requestExecutor, List<GetRequest> commands)
         {
-            _cache = cache;
+            _requestExecutor = requestExecutor;
+            _cache = _requestExecutor.Cache;
             _commands = commands;
             ResponseType = RavenCommandResponseType.Raw;
         }
 
         public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
         {
             _baseUrl = $"{node.Url}/databases/{node.Database}";
+            url = $"{_baseUrl}/multi_get";
+
+            var aggressiveCacheOptions = _requestExecutor.AggressiveCaching.Value;
+            if (aggressiveCacheOptions?.Mode == AggressiveCacheMode.TrackChanges)
+            {
+                Result = new List<GetResponse>();
+                foreach (var command in _commands)
+                {
+                    if(command.SkipAggressiveCache) break;
+                    var cacheKey = GetCacheKey(command, out string _);
+                    using (var cachedItem = _cache.Get(ctx, cacheKey, out _, out var cached))
+                    {
+                        if (cached == null || 
+                            cachedItem.Age > aggressiveCacheOptions.Duration ||
+                            cachedItem.MightHaveBeenModified)
+                            break;
+                        
```

Comment:
Double-check. Do we need to apply logic here from 'RequestExecutor.TryGetFromCache' ?

Reply:
That is where I took the logic from, yes.

Reply:
There was additional logic for delete, but it is not possible to have that case here, wanted to double-check.